### PR TITLE
Simplify UI for saving and publishing.

### PR DESCRIPTION
**Pull request by patbl**
See tog22/eaforum#62 for the motivation.

This would make adding more sub-forums much more difficult. (Less Wrong has Discussion and Main.) I'm not sure whether the trade-off is worthwhile. 

The volume on the EA forum is pretty low, and more-casual discussions happen on Facebook. If a need arose for sorting content, these changes (including removing tags) would make doing so more difficult.

But that that problem is hypothetical, and that the EA Forum will probably need to be replaced within a couple of years. So I'm OK with making decisions on that time horizon.

New-article UI:

draft-before-save

After clicking "Save as Draft":

draft-save-confirmation

After clicking "Publish":

published-confirmation

unpublish-ui

After clicking "Update":

updated-confirmation

After clicking "Unpublish":

unpublish-confirmation

#49 should be merged first.

### CODE REVIEW

File: r2/r2/controllers/front.py [hunk 1]
```diff
@@ -562,19 +564,21 @@ def GET_submit(self, can_submit, url, title, tags):
...
-            # An editor can save to the current subreddit irrspective of the original author's karma
-            subreddits = [sr for sr in Subreddit.submit_sr(c.user) if sr.is_editor(c.user)]
 
         captcha = Captcha(tabular=False) if c.user.needs_captcha() else None
-
-        return FormPage(_("Edit article"),
-                      show_sidebar = True,
-                      content=EditLink(article, subreddits=subreddits, tags=article.tag_names(), captcha=captcha)).render()
+        draft_subreddit = Subreddit.draft_subreddit(author)
+        main_subreddit = Subreddit.main_subreddit()
```
Comment (patbl) [inline r2/r2/controllers/front.py]: These two lines are duplicated. Not great, but whatevs.

File: r2/r2/models/subreddit.py [hunk 2]
```diff
@@ -399,6 +399,17 @@ def submit_sr(cls, user):
         srs.sort(key=lambda a:a.title)
         return srs
 
+    @classmethod
+    def main_subreddit(cls):
+        try:
+            return Subreddit._by_name("main")
+        except NotFound:
+            return Subreddit._by_name("lesswrong")
```
Comment (patbl) [inline r2/r2/models/subreddit.py]: I don't think you need this anymore.

File: r2/r2/models/subreddit.py [hunk 3]
```diff
@@ -399,6 +399,17 @@ def submit_sr(cls, user):
         srs.sort(key=lambda a:a.title)
         return srs
 
+    @classmethod
+    def main_subreddit(cls):
+        try:
+            return Subreddit._by_name("main")
```
Comment (patbl) [inline r2/r2/models/subreddit.py]: Can you use `cls`?

**Comment by peterhurford** [comment]
@patbl Do you publicly review your own PRs?

**Comment by peterhurford** [comment]
@patbl I like this UI. I don't think it would be too difficult to revert if we did need a multi-forum interface (which I think is unlikely).

**Pull request merged**